src/sentry/notifications/utils/__init__.py

```diff
@@ -200,9 +222,31 @@ def has_integrations(organization: Organization, project: Project) -> bool:
 
     project_plugins = plugins.for_project(project, version=1)
     organization_integrations = Integration.objects.filter(organizations=organization).first()
+    # TODO: fix because project_plugins is an iterator and thus always truthy
     return bool(project_plugins or organization_integrations)
 
 
+def is_alert_rule_integration(provider: IntegrationProvider) -> bool:
+    return any(feature == IntegrationFeatures.ALERT_RULE for feature in provider.features)
+    # return any(lambda x: x == IntegrationFeatures.ALERT_RULE, provider.features)
+
+
+def has_alert_integration(project: Project) -> bool:
+    org = project.organization
+
+    # check integrations
+    providers = filter(is_alert_rule_integration, list(integrations.all()))
+    provider_keys = map(lambda x: cast(str, x.key), providers)
+    if Integration.objects.filter(organizations=org, provider__in=provider_keys).exists():
+        return True
+
+    # check plugins
+    from sentry.plugins.base import plugins
+
+    project_plugins = plugins.for_project(project, version=None)
+    return any(plugin.get_plugin_type() == "notification" for plugin in project_plugins)
```

Comment:
are webhooks counted in this?

Reply:
@MeredithAnya Yup, it's a subclass of `NotificationPlugin`